ghost.py

```diff
@@ -986,6 +995,130 @@ def _get_document(self, key_name):
         return result['hits']['hits'] if result else {}
 
 
+class S3Storage(object):
+    def __init__(self,
+                 db_path=STORAGE_DEFAULT_PATH_MAPPING['s3'],
+                 aws_access_key_id=None or os.environ.get(
+                         'AWS_ACCESS_KEY_ID'),
+                 aws_secret_access_key=None or os.environ.get(
+                         'AWS_SECRET_ACCESS_KEY'),
+                 aws_session_token=None or os.environ.get('AWS_SESSION_TOKEN'),
+                 profile_name=None or os.environ.get('AWS_PROFILE'),
+                 region_name=None or os.environ.get('AWS_DEFAULT_REGION'),
+                 bucket_configuration=None):
+        """Initializes the storage client
+        :param db_path: The bucket name
+
+        Keyword arguments:
+        bucket_configuration
+        """
+        if not S3_EXISTS:
+            raise ImportError('boto3 and botocore must be installed first')
+        if not db_path:
+            raise GhostError('DB path (a bucket name) must be provided')
+        session = boto3.Session(
+            aws_access_key_id=aws_access_key_id,
+            aws_secret_access_key=aws_secret_access_key,
+            aws_session_token=aws_session_token,
+            profile_name=profile_name,
+            region_name=region_name)
+        self.client = session.client('s3')
+        self.db_path = db_path
+        if bucket_configuration:
+            self.bucket_configuration = bucket_configuration
+        else:
+            self.bucket_configuration = {'LocationConstraint': 'us-west-1'}
+
+    def init(self):
+        """Create a bucket.
+        """
+        self.client.create_bucket(
```

Comment:
It isn't. Only for US regions. I've added a `try` `except` clause to make it idempotent.